apiary.apib

```diff
@@ -4,7 +4,107 @@ FORMAT: 1A
 
 The API for interacting with IPFS nodes.
 
-# Group add
+# Group add [POST /add{?arg}{&recursive,quiet,silent,progress,trickle,only%2Dhash,wrap%2Dwith%2Ddirectory,hidden,chunker,pin}]
+Add an object to IPFS.
+
+Note that directories are added recursively, to form the ipfs
+MerkleDAG. A smarter partial add with a staging area (like git)
+remains to be implemented.
+
+#### curl
+
+    curl -F "file=@test/" -r http://localhost:5001/api/v0/add
```

Comment:
Our multipart messages changed recently (during 32c3) from nested to flat multipart, what this means is that:

OLD:
```
multipart message starts

folderA starts

  file1 inside folderA starts
    Jeromy likes liquorice 
  file1 inside folderA ends

  file2 inside folderA starts
    Jeromy ate all the liquorice 
  file2 inside folderA ends  

folder A ends

fileZ starts
  no more liquorice for Jeromy
fileZ ends

multipart message ends
```

NEW:
```
multipart message starts

file1 inside folderA starts
  Jeromy likes liquorice 
file1 inside folderA ends

file2 inside folderA starts
  Jeromy ate all the liquorice 
file2 inside folderA ends  

fileZ starts
  no more liquorice for Jeromy
fileZ ends

multipart message ends
```

Not sure if you are going to find a curl command (unless you create it) to do this.